manifests/templates/reconciler-manager-configmap.yaml

```diff
@@ -101,8 +101,8 @@ data:
                - NET_RAW
            imagePullPolicy: IfNotPresent
          - name: git-sync
-           image: gcr.io/config-management-release/git-sync:v3.6.9-gke.2__linux_amd64
-           args: ["--root=/repo/source", "--dest=rev", "--max-sync-failures=30", "--error-file=error.json"]
+           image: gcr.io/config-management-release/git-sync:v4.0.0-gke.1__linux_amd64
```

Comment:
nit: I think it may help to break this up into two changes:
1. refactor the env variables to the constants without changing functionality
2. upgrade git-sync and make associated changes to env vars

This would make it easier to visualize what changes come with the git-sync upgrade, and make for a smaller change in case we need to revert

Reply:
That's a good point. Extract the non-functional change into https://github.com/GoogleContainerTools/kpt-config-sync/pull/904.